packages/kbn-scout/src/playwright/page_objects/fiter_bar.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ScoutPage } from '../fixtures/types';
+import { expect } from '..';
+
+interface AddFilterOptions {
+  field: string;
+  operator: 'is' | 'is not' | 'is one of' | 'is not one of' | 'exists' | 'does not exist';
+  value: string;
+}
+
+interface FilterExistsOptions {
+  field: string;
+  value: string;
+  enabled?: boolean;
+  pinned?: boolean;
+  negated?: boolean;
+}
+
+export class FilterBar {
+  constructor(private readonly page: ScoutPage) {}
+
+  addFilter = async (options: AddFilterOptions) => {
+    await this.page.testSubj.click('addFilter');
+    await this.page.testSubj.waitForSelector('addFilterPopover');
+    // set field name
+    await this.page.testSubj.typeWithDelay(
+      'filterFieldSuggestionList > comboBoxSearchInput',
+      options.field
+    );
+    await this.page.click(`.euiFilterSelectItem[title="${options.field}"]`);
+    // set operator
+    await this.page.testSubj.typeWithDelay(
+      'filterOperatorList > comboBoxSearchInput',
+      options.operator
+    );
+    await this.page.click(`.euiFilterSelectItem[title="${options.operator}"]`);
+    // set value
+    await this.page.testSubj.locator('filterParams').locator('input').fill(options.value);
+    // save filter
+    await this.page.testSubj.click('saveFilter');
+
+    await expect(this.page.testSubj.locator('^filter-badge')).toBeVisible();
+  };
+
+  async hasFilter(options: FilterExistsOptions) {
```

Comment:
Is there a reason to mix styles in this file?
`addFilter = async (options: AddFilterOptions) => {` further up in the file and  `async hasFilter(options: FilterExistsOptions) {` here :thinking:

Reply:
Oh, I think I didn't word my question well. It was not about the different types for options but rather about the "arrow function" versus "classical function"  definition style.

Reply:
oh I see. Good point, applying changes for consistency

Reply:
done in d93ed528aa292c366f074db108cad0628b8259d3